flink-contrib/flink-storm/src/main/java/org/apache/flink/storm/api/FlinkClient.java

```diff
@@ -181,9 +188,11 @@ public void submitTopologyWithOpts(final String name, final String uploadedJarLo
 			throw new RuntimeException("Problem with jar file " + uploadedJarLocation, e);
 		}
 
-		/* set storm configuration */
-		if (this.conf != null) {
-			topology.getExecutionEnvironment().getConfig().setGlobalJobParameters(new StormConfig(this.conf));
+		try {
+			FlinkClient.addStormConfigToTopology(topology, conf);
+		} catch(ClassNotFoundException e) {
+			LOG.error("Could not register class for Kryo serialization.", e);
+			throw new InvalidTopologyException();
```

Comment:
Would be good to add the logged error message also to the exception. Gives more instant feedback to the user what exactly went wrong.